File: reverse.ts
import * as readline from 'readline';

function isAscendingDigits(num: number): boolean {
    const numStr = String(num);
    if (numStr.length <= 1) {
        return false; // Числа из одной цифры не считаются
    }

    for (let i = 1; i < numStr.length; i++) {
        if (numStr[i] <= numStr[i - 1]) {
            return false;
        }
    }
    return true;
}

function reverseNumber(num: number): number {
    const numStr = String(num);
    const reversedStr = numStr.split("").reverse().join("");
    try {
        return parseInt(reversedStr, 10);
    } catch (e: any) {
        console.error(`Превышение диапазона Int для числа ${num}`);
        return 0;
    }
}

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
});

rl.question("Введите последовательность чисел, разделенных пробелами:\n", (input: string) => {
    const numbersStr = input ? input.split(" ") : [];

    const numbers: number[] = numbersStr.map(Number).filter(num => !isNaN(num));

    const result = numbers.filter(isAscendingDigits).map(reverseNumber);

    let output = "Результат: ";
    if (result.length === 0) {
        output += "В последовательности нет чисел, соответствующих условию.";
    } else {
        output += result.join(" ");
    }

    console.log(output);
    rl.close();
});

File: reverse.rs
use std::io::{self, Write};

fn is_ascending_digits(num: i32) -> bool {
    let num_str = num.to_string();
    if num_str.len() <= 1 {
        return false;
    }

    for i in 1..num_str.len() {
        if num_str.chars().nth(i).unwrap() <= num_str.chars().nth(i - 1).unwrap() {
            return false;
        }
    }
    true
}

fn reverse_number(num: i32) -> i32 {
    let num_str = num.to_string();
    let reversed_str: String = num_str.chars().rev().collect();

    match reversed_str.parse::<i32>() {
        Ok(reversed_num) => reversed_num,
        Err(_) => {
            eprintln!("Превышение диапазона i32 для числа {}", num);
            0
        }
    }
}

fn main() -> io::Result<()> {
    print!("Введите последовательность чисел, разделенных пробелами: ");
    io::stdout().flush()?;  // Ensure prompt is displayed

    let mut input = String::new();
    io::stdin().read_line(&mut input)?;

    let numbers_str: Vec<&str> = input.trim().split_whitespace().collect();
    let mut numbers: Vec<i32> = Vec::new();

    for num_str in numbers_str {
        match num_str.parse::<i32>() {
            Ok(num) => numbers.push(num),
            Err(_) => {
                eprintln!("Некорректный ввод: {} не является числом.", num_str);
            }
        }
    }

    let mut result: Vec<i32> = Vec::new();

    for num in numbers {
        if is_ascending_digits(num) {
            result.push(reverse_number(num));
        }
    }

    print!("Результат: ");
    if result.is_empty() {
        println!("В последовательности нет чисел, соответствующих условию.");
    } else {
        for (i, num) in result.iter().enumerate() {
            print!("{}", num);
            if i < result.len() - 1 {
                print!(" ");
            }
        }
        println!();
    }

    Ok(())
}
